src/main/java/com/cdancy/jenkins/rest/filters/ScrubNullFolderParam.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cdancy.jenkins.rest.filters;
+
+import static com.cdancy.jenkins.rest.JenkinsConstants.OPTIONAL_FOLDER_PATH_PARAM;
+
+import javax.inject.Singleton;
+
+import org.jclouds.http.HttpException;
+import org.jclouds.http.HttpRequest;
+import org.jclouds.http.HttpRequestFilter;
+
+@Singleton
+public class ScrubNullFolderParam implements HttpRequestFilter {
+
+    private static final String SCRUB_NULL_PARAM = "/%7B" + OPTIONAL_FOLDER_PATH_PARAM + "%7D";
+    private static final String EMPTY_STRING = "";
+
+    @Override
+    public HttpRequest filter(final HttpRequest request) throws HttpException {        
+        final String requestPath = request.getEndpoint().getRawPath().replaceAll(SCRUB_NULL_PARAM, EMPTY_STRING);
+        return request.toBuilder().fromHttpRequest(request).replacePath(requestPath).build();
+    }
```

Comment:
@cdancy @Dhivyaa21 How should the user specify the optional path when calling the JobsApi methods?
1. Jenkins URL style: `job/folderA/job/folderB/job/jobName`
1. Simple path expression: `folderA/folderB/jobName`

This method looks like the right place to convert the simple path expression to the URL. It would make life easier for the user. What do you think? Should it be this method, or should it be in another implementation of `HttpRequestFilter`?

Reply:
Hmmm ... I suppose both are possible right? I'm open for suggestions here and yes this would be the best place IMO.

Reply:
Shouldn't be too bad to do something like: ensure each "name" is preceded by "job" and if not then inject into string.